### PR TITLE
added solution method

### PR DESCRIPTION
**Pull request by tsehelnyk**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/Cell.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.impl;
+
+public class Cell<K, V> {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/impl/Cell.java]: this class should be a private inner class

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,13 +3,41 @@
...
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int CELLS_LENGTH = 10;
+    private Cell[] cell = new Cell[CELLS_LENGTH];
+    private int freeCell = 0;
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i <= freeCell; i++) {
+            if (value != null) {
+                if (freeCell == 0) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: please, simplify this method

File: src/main/java/core/basesyntax/impl/Cell.java (hunk 1 above)
Comment (tsehelnyk) [reply src/main/java/core/basesyntax/impl/Cell.java]: ok)

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (tsehelnyk) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Done!

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -4,12 +4,62 @@
...
 public class StorageImpl<K, V> implements Storage<K, V> {
 
+    private static final int CELLS_LENGTH = 10;
+    private Cell[] cell = new Cell[CELLS_LENGTH];
+    private int freeCell = 0;
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i <= freeCell; i++) {
+            if (value != null && freeCell == 0) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: there are a lot of redundant checks. please, simplify

**Pull request closed without merging**